File: alloy-gem5/src/brg-utst/test-scratchpad/TestScratchpadHarness.py
from TestHarness import TestHarness
from m5.params import *
from m5.proxy import *

class TestScratchpadHarness(TestHarness):
  type = 'TestScratchpadHarness'
  cxx_header = "brg-utst/test-scratchpad/test_scratchpad_harness.hh"

File: alloy-gem5/src/brg-utst/test-scratchpad/test_scratchpad_harness.cc
//========================================================================
// Test Scratchpad Harness
//========================================================================
//
// Authors  : Tuan Ta
// Date     : July 2, 2019
//

#include "test_scratchpad_harness.hh"
#include "test_scratchpad_suite.hh"

TestScratchpadHarness::TestScratchpadHarness(const Params* params)
    : TestHarness(params)
{
  test_suite_ptr = new TestScratchpadSuite(num_test_ports);

  // init directed tests
  test_suite_ptr->initDirectedTests();
}

TestScratchpadHarness*
TestScratchpadHarnessParams::create()
{
  return new TestScratchpadHarness(this);
}
